tests/test_request_overrider.js

```diff
@@ -522,8 +528,11 @@ test('socket has ref() and unref() method', t => {
 
   const req = http.get('http://example.test')
   req.once('socket', socket => {
-    socket.ref()
-    socket.unref()
+    expect(socket)
+      .to.respondTo('ref')
+      .and.to.to.respondTo('unref')
+    expect(socket.ref()).to.be.undefined()
+    expect(socket.unref()).to.be.undefined()
```

Comment:
I know this is the expected behavior, but this is a bug in our socket implementation. Most of its methods should return `this`. Out of scope of this PR, but should probably be fixed at some point.

Reply:
Ahh, good to know. I'll add a todo comment.

Without these tests these functions weren't called and I felt better about asserting their return value than just calling them… (to imply what exactly?) but I'm glad it surfaced your comment.